mesonbuild/options.py

```diff
@@ -1407,6 +1402,7 @@ def initialize_from_subproject_call(self,
 
     def update_project_options(self, project_options: MutableKeyedOptionDictType, subproject: SubProject) -> None:
         for key, value in project_options.items():
+            assert key.machine == MachineChoice.HOST
```

Comment:
Prefer `is` and `is not` with Enum values, since `==` just becomes an extra level of lookup for `is` in this case.